sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersClientBuilder.java

```diff
@@ -257,7 +272,7 @@ UserAgentPolicy createUserAgentPolicy(
     }
 
     RetryPolicy createRetryPolicy() {
-        return new RetryPolicy();
+        return this.retryPolicy == null ? new RetryPolicy() : this.retryPolicy;
```

Comment:
Why we need this method? Tehnically, it is not always createRetryPolicy().

Reply:
Yea good point. It is not necessary to have this logic be in a separate method, I will move back